docs/content/concepts/dagster-pipes/pyspark.mdx

```diff
@@ -0,0 +1,328 @@
+
+     Make sure `hadoop-aws` JAR and AWS Java SDK bundle versions match your Spark/Hadoop build.
+
+- We are going to upload the PySpark script to the S3 bucket directly inside the Dagster asset. In production, consider doing this via CI/CD instead.
+
+---
+
+## Step 1: Writing the Dagster orchestration code
+
+We will set up a few non-default Pipes components to streamline the otherwise challenging problem of orchestrating Spark jobs.
+
+1. Let's start by creating the asset and opening a Pipes session. We will be using S3 to pass Pipes messages from the Spark job to Dagster, so we will create `PipesS3MessageReader` and `PipesS3ContextInjector` objects. (Technically, it's not strictly required to use S3 for passing the Dagster context, but storing it there will decrease the CLI arguments size.)
+
+```python file=/guides/dagster/dagster_pipes/pyspark/dagster_code.py startafter=start_pipes_session_marker endbefore=end_pipes_session_marker
+import os
+import subprocess
+from pathlib import Path
+from typing import Mapping, Sequence
+
+import boto3
+from dagster_aws.pipes import PipesS3ContextInjector, PipesS3MessageReader
+
+import dagster as dg
+
+LOCAL_SCRIPT_PATH = Path(__file__).parent / "script.py"
+
+
+@dg.asset
+def pipes_spark_asset(context: dg.AssetExecutionContext):
+    s3_client = boto3.client("s3")
+
+    bucket = os.environ["DAGSTER_PIPES_BUCKET"]
+
+    # upload the script to S3
+    # ideally, this should be done via CI/CD processes and not in the asset body
+    # but for the sake of this example we are doing it here
+    s3_script_path = f"{context.dagster_run.run_id}/pyspark_script.py"
+    s3_client.upload_file(LOCAL_SCRIPT_PATH, bucket, s3_script_path)
+
+    context_injector = PipesS3ContextInjector(
+        client=s3_client,
+        bucket=bucket,
+    )
+
+    message_reader = PipesS3MessageReader(
+        client=s3_client,
+        bucket=bucket,
+        # the following setting will configure the Spark job to collect logs from the driver
+        # and send them to Dagster via Pipes
+        include_stdio_in_messages=True,
+    )
+
+    with dg.open_pipes_session(
+        context=context,
+        message_reader=message_reader,
+        context_injector=context_injector,
+    ) as session:
+```
+
+Notice how `PipesS3MessageReader` has `include_stdio_in_messages=True`. This setting will configure the Pipes message writer in the Spark job to collect logs from the Spark driver and send them to Dagster via Pipes messages.
+
+2. We will be using CLI arguments to pass the bootstrap information from Dagster to the Spark job. In other Pipes workflows this is typically done via environment variables, but setting them for Spark jobs can be complicated (the exact way of doing this depends on the Spark deployment) or not possible at all. CLI arguments are a convenient alternative. We will fetch them from the `session.get_bootstrap_cli_arguments` method. We pass these arguments to `spark-submit` along with a few other settings.
```

Comment:
```suggestion
2. We will be using CLI arguments to pass the bootstrap information from Dagster to the Spark job. In other Pipes workflows, this is typically done via environment variables, but setting them for Spark jobs can be complicated (the exact way of doing this depends on the Spark deployment) or not possible at all. CLI arguments are a convenient alternative. We will fetch them from the `session.get_bootstrap_cli_arguments` method. We pass these arguments to `spark-submit` along with a few other settings.
```

Reply:
I would reorganize this a bit:

```
We will be using CLI arguments that we fetch from `session.get_bootstrap_cli_arguments` and store in `dagster_pipes_args`. We pass these arguments to `spark-submit`, along with a few other settings, in order to pass the bootstrap information from Dagster to the Spark job.
```

I would move the sentences about environment variables to a note below this step (see my next comment).